sys/kern/vm_map.c

```diff
@@ -328,8 +330,15 @@ vm_map_t *vm_map_clone(vm_map_t *map) {
   WITH_MTX_LOCK (&map->mtx) {
     vm_segment_t *it;
     TAILQ_FOREACH (it, &map->entries, link) {
-      vm_object_t *obj = vm_object_clone(it->object);
-      vm_segment_t *seg = vm_segment_alloc(obj, it->start, it->end, it->prot);
+      vm_object_t *obj;
+      vm_segment_t *seg;
+      if (it->flags & VM_SHARED) {
+        refcnt_acquire(&it->object->ref_counter);
+        seg = it;
+      } else {
+        obj = vm_object_clone(it->object);
```

Comment:
It's not clear for me what should be done with `ref_counter` in `vm_object_clone`.
Is it possible to have `ref_counter` different than `1` here?
Could you explain?

Reply:
vm_object_clone clones only the data of the vm_object. That means ref_counter will be equal to 1, so it's not possible to have ref_counter different than 1 here